### PR TITLE
New Alpha -- Contingent Claims Analysis Alpha

### PR DESCRIPTION
**Pull request by simonsonjack**
Created new Alpha

#### Description
New Alpha for open source to help instruct the community

#### Motivation and Context
Provide new Alpha demo for the community

#### Requires Documentation Change
No

#### How Has This Been Tested?
Tested locally and in cloud. Will be reviewed by senior engineers before deployment.

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Non-functional change (xml comments/documentation/etc)

#### Checklist:

- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [x] I have added tests to cover my changes. 
- [x] All new and existing tests passed.
- [x] My branch follows the naming convention `bug-<issue#>-<description` or `feature-<issue#>-<description>`

### CODE REVIEW

File: Algorithm.Python/Alphas/ContingentClaimsAnalysisDefaultPredictionAlpha.py [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+            self.month = self.Time.month
+
+            ## Sort by dollar volume, lowest to highest
+            sortedByDollarVolume = sorted(coarse, key=lambda x: x.DollarVolume, reverse=True)
+
+            ## Filter for assets with fundamental data
+            filtered = [ x.Symbol for x in sortedByDollarVolume ]
+        
+            ## Return smallest 750 -- idea is that smaller companies are most likely to go bankrupt than blue-chip companies
+            self.symbols = filtered[:750]
```
Comment (AlexCatarino) [inline Algorithm.Python/Alphas/ContingentClaimsAnalysisDefaultPredictionAlpha.py]: The `FineFundamentalUniverseSelectionModel` will automatically remove the Coarse Fundamental that `HadFundamentalData` property is false. However, if we want to ensure that 750 symbols arrive at Fine Fundamental selection function, we need to filter in this method (before line 81).

File: Algorithm.Python/Alphas/ContingentClaimsAnalysisDefaultPredictionAlpha.py [hunk 2]
```diff
@@ -0,0 +1,229 @@
+''' Contingent Claim Analysis is put forth by Robert Merton, recepient of the Noble Prize in Economics in 1997 for his work in contributing to
```
Comment (AlexCatarino) [inline Algorithm.Python/Alphas/ContingentClaimsAnalysisDefaultPredictionAlpha.py]: Need to add QuantConnect's header.

File: Algorithm.Python/Alphas/ContingentClaimsAnalysisDefaultPredictionAlpha.py [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+
+
+class ContingentClaimsAnalysisAlphaModel:
+    
+    def __init__(self, *args, **kwargs):
+        self.symbolDataBySymbol = {}
+        self.month = None
+        self.default_threshold = kwargs['default_threshold'] if 'default_threshold' in kwargs else 0.25
+        self.expiry = None
+        self.epsilon = kwargs['epsilon'] if 'epsilon' in kwargs else 0.00001     ## This serves as a check to filter out symbols with a default probability of, e.g., 2.89e-20
```
Comment (AlexCatarino) [inline Algorithm.Python/Alphas/ContingentClaimsAnalysisDefaultPredictionAlpha.py]: `self.expiry` and `self.epsilon` are not being used.

File: Algorithm.Python/Alphas/ContingentClaimsAnalysisDefaultPredictionAlpha.py (hunk 1 above)
Comment (simonsonjack) [reply Algorithm.Python/Alphas/ContingentClaimsAnalysisDefaultPredictionAlpha.py]: Not sure I follow we return all symbols from Coarse and then take 750 after fineFilter has been created?

**Pull request merged**